homeassistant/components/aruba/device_tracker.py

```diff
@@ -46,9 +54,43 @@ def __init__(self, config):
         self.username = config[CONF_USERNAME]
         self.password = config[CONF_PASSWORD]
 
+        global USE_API
+        global login_result_sid
+
         self.last_results = {}
 
-        # Test the router is accessible.
+        # Test if we can use REST API
+        # Aruba Instant API only likes double quotes in the json data and rejects single quotes
+        login_data = (
+            '{"user": "' + self.username + '", "passwd": "' + self.password + '" }'
+        )
+        login_headers = {"Content-Type": "application/json"}
+        login_method = "POST"
+        login_resource = "https://" + self.host + ":4343/rest/login"
+
+        rest_login = RestData(
+            login_method,
+            login_resource,
+            login_headers,
+            login_data,
+            False,
+            DEFAULT_TIMEOUT,
+        )
+        rest_login.update()
+
+        if rest_login.headers is not None:
+            login_result_content_type = rest_login.headers.get("content-type")
+            if login_result_content_type == "application/json":
+                login_result_json = json.loads(rest_login.data)
+                login_result_status = login_result_json.get("Status")
+                if login_result_status == "Success":
+                    USE_API = True
+                    login_result_sid = login_result_json.get("sid")
+                    _LOGGER.debug("Successfully logged in to Aruba Instant REST API")
+                else:
+                    USE_API = DEFAULT_USE_API
+                    _LOGGER.debug("Error logging in to Aruba Instant REST API")
```

Comment:
All communication to external devices or services must be wrapped in an external Python library hosted on PyPi.

<https://developers.home-assistant.io/docs/development_checklist>